### PR TITLE
Split L1 graph-construction tests by domain

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary

- replace the 7,453-line L1 graph-construction monolith with six cohesive domain modules under `tests/build_graph/`
- keep vision-language/multimodal separate from speech/ASR/TTS/diarization coverage
- move shared checker helpers and stable specialized-model inventories into non-collected `tests/build_graph/_support.py`
- update CI, dashboard scanning, affected-model regressions, active docs, Copilot instructions, and tracked project skills to invoke `tests/build_graph`
- validate inferred output dtypes through `onnx_ir.Value.dtype`

## Layout

- `tests/build_graph/core_test.py`
- `tests/build_graph/cache_test.py`
- `tests/build_graph/diffusion_audio_test.py`
- `tests/build_graph/recurrent_test.py`
- `tests/build_graph/speech_test.py`
- `tests/build_graph/vision_language_test.py`
- `tests/build_graph/_support.py` (not collected)

## Behavior preservation

- old collection: 1,635 tests
- new collection: 1,635 tests
- normalized node IDs: identical after stripping only the collected file path prefix
- all 50 moved test class ASTs are identical; marker, parameter, skip, `--fast`, and `--models` behavior is preserved

## Validation

- `python -m pytest tests/build_graph -q --tb=short -n auto` (1,583 passed, 52 skipped)
- `python -m pytest tests/build_graph --fast -q --tb=short -n auto` (710 passed, 925 skipped)
- `python -m pytest tests/build_graph --models qwen2 -q --tb=short -n auto` (226 passed, 1,409 skipped)
- focused helper, UNet, dashboard, and affected-model tests (159 passed)
- current CLI and native-audio regression tests (151 passed)
- repository-wide stale L1 path/import search
- lintrunner on changed files
- `git diff --check`
- exact-head GPT-5.6 Sol functional regression review

### CODE REVIEW

File: tests/_build_graph_support.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+        _shape_inference(model)
+        for output in model.graph.outputs:
+            assert output.shape is not None, (
+                f"{model_type}/{sub_name}: output '{output.name}' "
+                f"has no shape after shape inference"
+            )
+            assert output.type is not None, (
+                f"{model_type}/{sub_name}: output '{output.name}' "
+                f"has no dtype after shape inference"
+            )
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/_build_graph_support.py]: Fixed in `4adc5b90`. The helper now validates `onnx_ir.Value.dtype` after shape inference. Its existing parametrized output-metadata tests exercise the helper; I also ran the focused Qwen2 metadata case and the complete 1,635-test L1 suite.

File: tests/build_graph_diffusion_audio_test.py [hunk 2]
```diff
@@ -0,0 +1,1166 @@
...
+
+        out = sess.run({"input_values": waveform, "attention_mask": attention_mask})
+        sess.close()
+
+        logits = out["logits"]
+        assert logits.shape[0] == 1  # batch
+        assert logits.shape[1] > 0  # num_frames (after CNN downsampling)
+        assert logits.shape[2] == config.vocab_size  # CTC vocab
+
+    """Verify UNet2DConditionModel graph construction."""
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/build_graph_diffusion_audio_test.py]: Fixed in `4adc5b90`. The standalone class-body string is now a section comment, preserving the existing `TestBuildMMSGraph` node IDs and class boundary while removing the inert expression.

File: tests/build_graph/core_test.py [hunk 3]
```diff
@@ -0,0 +1,893 @@
...
+
+_MODEL_CONFIGS: list[tuple[str, dict]] = [(mt, ov) for mt, ov, _ in ALL_CAUSAL_LM_CONFIGS]
+_MODEL_PARAMS = _make_params(ALL_CAUSAL_LM_CONFIGS)
+_ENCODER_MODEL_CONFIGS: list[tuple[str, dict]] = [(mt, ov) for mt, ov, _ in ENCODER_CONFIGS]
+_ENCODER_MODEL_PARAMS = _make_params(ENCODER_CONFIGS)
+_SEQ2SEQ_MODEL_CONFIGS: list[tuple[str, dict]] = [(mt, ov) for mt, ov, _ in SEQ2SEQ_CONFIGS]
+_SEQ2SEQ_MODEL_PARAMS = _make_params(SEQ2SEQ_CONFIGS)
+_VISION_MODEL_CONFIGS: list[tuple[str, dict]] = [(mt, ov) for mt, ov, _ in VISION_CONFIGS]
+_VISION_MODEL_PARAMS = _make_params(VISION_CONFIGS)
+_DETECTION_MODEL_CONFIGS: list[tuple[str, dict]] = [
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/build_graph/core_test.py]: Fixed in `4e3a5e90`. Repo-wide search confirmed `_DETECTION_MODEL_CONFIGS` was genuinely unused, so I removed it while leaving `_DETECTION_MODEL_PARAMS = _make_params(DETECTION_CONFIGS)` unchanged.

File: tests/build_graph/core_test.py [hunk 4]
```diff
@@ -0,0 +1,893 @@
...
+
+from mobius._builder import DTYPE_MAP, build_from_module
+from mobius._configs import AudioConfig, VisionConfig
+from mobius._registry import registry
+from mobius.integrations.transformers._config_resolver import _default_task_for_model
+from mobius.tasks import CausalLMTask, Phi4MMMultiModalTask, get_task
+
+_MODEL_CONFIGS: list[tuple[str, dict]] = [(mt, ov) for mt, ov, _ in ALL_CAUSAL_LM_CONFIGS]
+_MODEL_PARAMS = _make_params(ALL_CAUSAL_LM_CONFIGS)
+_ENCODER_MODEL_CONFIGS: list[tuple[str, dict]] = [(mt, ov) for mt, ov, _ in ENCODER_CONFIGS]
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/build_graph/core_test.py]: Fixed in `4e3a5e90`. Repo-wide search confirmed `_ENCODER_MODEL_CONFIGS` was genuinely unused, so I removed it while leaving `_ENCODER_MODEL_PARAMS = _make_params(ENCODER_CONFIGS)` unchanged.

File: tests/build_graph/core_test.py [hunk 5]
```diff
@@ -0,0 +1,893 @@
...
+_ENCODER_MODEL_PARAMS = _make_params(ENCODER_CONFIGS)
+_SEQ2SEQ_MODEL_CONFIGS: list[tuple[str, dict]] = [(mt, ov) for mt, ov, _ in SEQ2SEQ_CONFIGS]
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/build_graph/core_test.py]: Fixed in `4e3a5e90`. Repo-wide search confirmed `_SEQ2SEQ_MODEL_CONFIGS` was genuinely unused, so I removed it while leaving `_SEQ2SEQ_MODEL_PARAMS = _make_params(SEQ2SEQ_CONFIGS)` unchanged.

File: tests/build_graph/_support.py [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+    # EAGLE-3 drafter: bespoke IO contract (inputs_embeds, fused_hidden,
+    # recycled_hidden and draft-vocab logits). Covered by _eagle3_test.py.
+    "Eagle3LlamaForCausalLM",
+    "LlamaForCausalLMEagle3",
+    "Eagle3Speculator",
+    "Eagle3DraftModel",
+}
+
+
+_KNOWN_UNTESTED_MODEL_TYPES: set[str] = {
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/build_graph/_support.py:267]: Fixed in `4e3a5e90`. I rejected the proposed `_make_params` filtering because known-untested aliases are an accounting allowlist, not test parameters. `known_untested_model_types()` now exposes the read-only inventory to registry completeness, preserving the existing two-alias semantics and the exact 1,635-node collection.

File: tests/build_graph/_support.py [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+            idx = seen.get(model_type, 0)
+            seen[model_type] = idx + 1
+            test_id = _SEMANTIC_IDS.get((model_type, idx), f"{model_type}_{idx}")
+        else:
+            test_id = model_type
+        params.append(pytest.param(model_type, overrides, id=test_id))
+    return params
+
+
+_SPECIALIZED_TEST_MODEL_TYPES: set[str] = {
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/build_graph/_support.py:130]: Fixed in `4e3a5e90`. I rejected the proposed `_make_params` filtering because it would change the proven collection contract. This inventory is not dead: it feeds both registry completeness and dashboard L1 coverage. `specialized_test_model_types()` now exposes a read-only frozenset to both real consumers, making the contract explicit without changing parametrization.

File: tests/build_graph/core_test.py [hunk 8]
```diff
@@ -0,0 +1,893 @@
...
+_SEQ2SEQ_MODEL_PARAMS = _make_params(SEQ2SEQ_CONFIGS)
+_VISION_MODEL_CONFIGS: list[tuple[str, dict]] = [(mt, ov) for mt, ov, _ in VISION_CONFIGS]
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/build_graph/core_test.py]: Fixed in `4e3a5e90`. Repo-wide search confirmed `_VISION_MODEL_CONFIGS` was genuinely unused, so I removed it while leaving `_VISION_MODEL_PARAMS = _make_params(VISION_CONFIGS)` unchanged.

File: tests/build_graph/_support.py [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/build_graph/_support.py:130]: Fixed in `4e3a5e90`. This inventory is a real input to registry completeness and dashboard L1 coverage, so removing or parameterizing it would change semantics. `specialized_test_model_types()` now consumes the backing set in-module and exposes an immutable view to both real consumers.

File: tests/build_graph/_support.py [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+    # EAGLE-3 drafter: bespoke IO contract (inputs_embeds, fused_hidden,
+    # recycled_hidden and draft-vocab logits). Covered by _eagle3_test.py.
+    "Eagle3LlamaForCausalLM",
+    "LlamaForCausalLMEagle3",
+    "Eagle3Speculator",
+    "Eagle3DraftModel",
+}
+
+
+_KNOWN_UNTESTED_MODEL_TYPES: set[str] = {
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/build_graph/_support.py:267]: Fixed in `4e3a5e90`. This is not dead code: the two-entry allowlist is required by the registry-completeness accounting test. `known_untested_model_types()` now consumes the backing set in-module and exposes an immutable view to that consumer without affecting parametrization.
